docs/hello_nextflow/04_hello_science.md

```diff
@@ -0,0 +1,763 @@
+[88/1783aa] GATK_HAPLOTYPECALLER (2) | 3 of 3 ✔
+```
+
+If you'd like, you can use `.view()` again to peek at what the contents of the `` output channel looks like:
+
+```groovy title="hello-gatk.nf" linenums="92"
+SAMTOOLS_INDEX.out.view()
+```
+
+You'll see the channel contains the three expected tuples (filepaths truncated for readability).
+
+```console title="Output"
+[.../4c/e16099*/reads_son.bam, .../4c/e16099*/reads_son.bam.bai]
+[.../42/e70b8b*/reads_father.bam, .../42/e70b8b*/reads_father.bam.bai]
+[.../18/23b4bb*/reads_mother.bam, .../18/23b4bb*/reads_mother.bam.bai]
+```
+
+That will be much safer, going forward.
+
+### Takeaway
+
+You know how to make your workflow run on multiple samples (independently).
+
+### What's next?
+
+Make it easier to handle samples in bulk.
+
+---
+
+## 4. Make the workflow accept a text file containing a batch of input files
+
+A very common way to provide multiple data input files to a workflow is to do it with a text file containing the file paths. It can be as simple as a text list with one file path per line and nothing else, or the file can contain additional metadata, in which case it's often called a samplesheet.
+
+Here we are going to show you how to do the simple case.
+
+### 4.1. Examine the provided text file listing the input file paths
+
+We already made a text file listing the input file paths, called `sample_bams.txt`, which you can find in the `data/` directory.
+
+```txt title="sample_bams.txt"
+/workspace/gitpod/hello-nextflow/data/bam/reads_mother.bam
+/workspace/gitpod/hello-nextflow/data/bam/reads_father.bam
+/workspace/gitpod/hello-nextflow/data/bam/reads_son.bam
+```
+
+As you can see, we listed one file path per line, and they are absolute paths.
+
+!!! note
+The files we are using here are just on your Gitpod's local filesystem, but we could also point to files in cloud storage.
+
+### 4.2. Update the parameter default
+
+Let's switch the default value for our `reads_bam` input parameter to point to the `sample_bams.txt` file.
+
+_Before:_
+
+```groovy title="hello-gatk.nf" linenums="7"
+// Primary input
+params.reads_bam = [
+    "${projectDir}/data/bam/reads_mother.bam",
+    "${projectDir}/data/bam/reads_father.bam",
+    "${projectDir}/data/bam/reads_son.bam"
+]
+```
+
+_After:_
+
+```groovy title="hello-gatk.nf" linenums="7"
+// Primary input (list of input files, one per line)
+params.reads_bam = "${projectDir}/data/sample_bams.txt"
+```
+
+This way we can continue to be lazy, but the list of files no longer lives in the workflow code itself, which is a big step in the right direction.
+
+### 4.3. Update the channel constructor to read lines from a file
+
+Currently, our input channel constructor treats any files we give it as the data inputs we want to feed to the indexing process. Since we're now giving it a file that lists input file paths, we need to change its behavior to parse the file and treat the file paths it contains as the data inputs.
```

Comment:
```suggestion
Currently, our input channel constructor treats any files we give it as the data inputs we want to feed to the indexing process.
Since we're now giving it a file that lists input file paths, we need to change its behavior to parse the file and treat the file paths it contains as the data inputs.
```